libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,144 @@
+        /// <param name="userState">A <see cref="UserState"/> implementation.</param>
+        public AdaptiveDialogBot(
+            ResourceExplorer resourceExplorer,
+            string rootDialogId,
+            string defaultLocale,
+            ILogger<AdaptiveDialogBot> logger,
+            BotFrameworkAuthentication botFrameworkAuthentication,
+            SkillConversationIdFactoryBase skillConversationIdFactoryBase,
+            ConversationState conversationState,
+            UserState userState)
+        {
+            _resourceExplorer = resourceExplorer;
+            _rootDialogId = rootDialogId ?? "Main.dialog";
+            _defaultLocale = defaultLocale ?? DefaultLocale;
+            _logger = logger ?? NullLogger<AdaptiveDialogBot>.Instance;
+            _botFrameworkAuthentication = botFrameworkAuthentication ?? BotFrameworkAuthenticationFactory.Create();
+
+            _skillConversationIdFactoryBase = skillConversationIdFactoryBase;
+            _conversationState = conversationState;
+            _userState = userState;
+        }
+
+        async Task IBot.OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"IBot.OnTurnContext for dialog '{_rootDialogId}'");
+
+            using (var botFrameworkClient = _botFrameworkAuthentication.CreateBotFrameworkClient())
+            {
+                // Set up the TurnState the Dialog is expecting
+                SetUpTurnState(turnContext, botFrameworkClient);
+
+                // Load the Dialog from the ResourceExplorer
+                var rootDialog = await CreateDialogAsync(cancellationToken).ConfigureAwait(false);
+
+                // Run the Dialog
+                await rootDialog.RunAsync(turnContext, turnContext.TurnState.Get<ConversationState>().CreateProperty<DialogState>("DialogState"), cancellationToken).ConfigureAwait(false);
+
+                // Save any updates that have been made
+                await turnContext.TurnState.Get<ConversationState>().SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+                await turnContext.TurnState.Get<UserState>().SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private void SetUpTurnState(ITurnContext turnContext, BotFrameworkClient botFrameworkClient)
+        {
+            turnContext.TurnState.Add(botFrameworkClient);
+            turnContext.TurnState.Add(_skillConversationIdFactoryBase);
+            turnContext.TurnState.Add(_conversationState);
+            turnContext.TurnState.Add(_userState);
+            turnContext.TurnState.Add(_resourceExplorer);
+            turnContext.TurnState.Add(_resourceExplorer.TryGetResource(DefaultLg, out var resource) ? (LanguageGenerator)new ResourceMultiLanguageGenerator(DefaultLg) : new TemplateEngineLanguageGenerator());
```

Comment:
I'm not sure I understand this line of code, shouldn't I add "resource" to turn state if TryGetResource succeeds instead of constructing a new instance of ResourceMultiLanguageGenerator and then cast it to LanguageGenerator before adding it?